server/client/stress_test.go

```diff
@@ -0,0 +1,215 @@
+package client
+
+import (
+	"testing"
+	"os"
+	"github.com/elastic/hey-apm/server/api"
+
+	"sync"
+	"fmt"
+	"time"
+	"github.com/stretchr/testify/assert"
+	"net/http"
+	"github.com/pkg/errors"
+	"path/filepath"
+)
+
+var once sync.Once
+var environment *evalEnvironment
+
+func setupEnv() (*evalEnvironment, []string, error) {
+	once.Do(func() {
+		url := os.Getenv("ELASTICSEARCH_URL")
+		usr := os.Getenv("ELASTICSEARCH_USR")
+		pwd := os.Getenv("ELASTICSEARCH_PWD")
+		environment = NewEvalEnvironment("")
+		_, environment.es = elasticSearchUse("", url, usr, pwd)
+		apmDir := filepath.Join(os.Getenv("GOPATH"), "/src/github.com/elastic/apm-server")
+		_, environment.apm = apmSwitch(os.Stdout, apmDir, "master", "", []string{"c", "m", "u"})
+	})
+
+	flags := apmFlags(*environment.es, environment.apm.Url(), []string{"-e", "-E", "apm-server.shutdown_timeout=1s"})
+	err := apmStop(environment.apm)
+	if err == nil {
+		time.Sleep(time.Second * 5)
```

Comment:
What is this sleep for ?

Reply:
for a reason I don't understand, without it sometimes it hangs during startup... maybe an ungraceful TCP socket close from the previous run?  
without `time.Sleep` it might say "address already in use" (which makes sense given the 1sec shutdown timeout in the server), which makes sense; but with a 3 secs sleep sometimes just hangs which i don't understand. any hints?